python/nvfuser/__init__.py

```diff
@@ -310,11 +358,13 @@ def execute(
                     self.schedule()
                     self._finalize_schedule(inputs)
 
+        # TODO(crcrpar): Think about deprecating `self.fake_inputs` in favor of `InputDescriptor`.
```

Comment:
What are the tradeoffs between using `self.fake_inputs` and `self.input_descriptors`?